### DISCUSSION TITLE
per-app force_zero_scaling = false setting (HiDPI GTK2 issues)

### DISCUSSION THREAD

**Discussion by EmpathyZeroed** [hyprwm/Hyprland#8539] [Q&A] [answered]
I've had to set `force_zero_scaling = false` for Electron and QT apps HiDPI compatibility, but this causes GTK2 apps to appear too small (`GDK_SCALE` setting is ignored for some reason). If I set `force_zero_scaling = false` they become clear and sharply scaled but this breaks Electron and makes it blurry

Is it possible to `force_zero_scaling = false` only for a specified app? Window rules doesn't work. I want force_zero_scaling for xwayland be always true except for Pidgin (GTK2), smth like `windowrule = scale 2, class: Pidgin`

relevant snippet of my current hyprland.conf:
```
xwayland {
  force_zero_scaling = true
}

env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland

# QT comp
env = QT_QPA_PLATFORM,wayland;xcb
env = QT_AUTO_SCREEN_SCALE_FACTOR,1

# GTK comp
env = GDK_SCALE,2
```

**Comment by vaxerski** [comment]
not possible, due to how xwayland works.

**Reply by EmpathyZeroed** [reply]
could you please propose any workaround?

**Reply by vaxerski** [reply, marked as answer]
there is no workaround
